### PR TITLE
Custom attributes / json import & export

### PR DESCRIPTION
**Pull request by stephentuso**
Not sure if you will want to merge this, these are changes I need for a personal project and so it may not be very useful to others - I am generating an interactive image map from json, and I needed an easy way to export map data as json. I also needed more properties than just href, alt, and title so I added the ability to add/import custom attributes. The commit messages should explain the other changes I made.

Also, there are conflicts with merging to master because I started from the gh-pages branch, I had issues with the current master branch (couldn't create polygons).

### CODE REVIEW

File: index.htm [hunk 1]
```diff
@@ -58,12 +59,14 @@ <h1 id="logo">
...
-	<div id="code_content"></div>
+	<div>
+		<textarea id="code_content" onclick="this.focus();this.select();" readonly></textarea>
+	</div>
 </div>
 
 <!-- Edit details block -->
 <form id="edit_details">
-	<h5>Attrubutes</h5>
+	<h5>Attributes</h5>
```
Comment (summerstyle) [inline index.htm]: +

File: scripts.js [hunk 2]
```diff
@@ -128,7 +128,7 @@ function SummerHtmlImageMapCreator() {
 		extend : function(obj, options) {
 			var target = {};
 			
-			for (name in obj) {
+			for (var name in obj) {
```
Comment (summerstyle) [inline scripts.js]: +

File: scripts.js [hunk 3]
```diff
@@ -707,8 +786,41 @@ function SummerHtmlImageMapCreator() {
...
+			},
+			hide: function() {
+				utils.hide(block);
+			}
+		};
+	})();
+
+	//TODO (stephentuso): DRY
+	/* For json of created map */
+	var codeJson = (function(){
```
Comment (summerstyle) [inline scripts.js]: this is good idea, export and import to json

**Comment by summerstyle** [comment]
Hi, 
thank you for your pull-request. 
I created this project 4 years ago, and my old code is very bad.
Now, before I will add some features to editor, I am making big refactoring of this code. 
And, after this, I will divide interface parts from editor (https://github.com/summerstyle/app.js), and work with areas - to separate library too.
Some commits is alredy in master branch (you are right, there are bug with polygons), and other changes I don't push to master yet. And I can't merge your pull-request because code of gh-pages is old. 
But I fixed some bugs you finded, and  I will add import/export to json (this your idea is very good).

**Comment by stephentuso** [comment]
Looking back at old code is always fun. 
That makes sense, I wasn't really expecting it to get merged.
Sounds good, the main thing I added besides json import/export was the ability to add attributes other than `href`, `alt`, and `title`, you may want to consider adding that.

**Pull request closed without merging**